src/tmx/TmxUtils/test/TelemetrySerializerTest.cpp

```diff
@@ -0,0 +1,139 @@
+#include <gtest/gtest.h>
+#include "TelemetrySerializer.h"
+
+
+using namespace tmx::utils::telemetry;
+namespace unit_test{
+    class TelemetrySerializerTest: public ::testing::Test
+    {
+    public:
+        shared_ptr<PluginTelemetry> telemetryPointer;
+        shared_ptr<PluginTelemetry> telemetryPluginInfoOnlyPointer;
+        
+        ~TelemetrySerializerTest() override {};
+        
+        void SetUp() override{
+            PluginInfo pluginInfo = {
+                "1", //id
+                "CARMACloudPlugin", //name
+                "CARMA cloud plugin for making websocket connection with CARMA cloud.", //description
+                "7.6.0", //version
+            };
+
+            PluginInstallation installation = {
+                0, //enabled
+                "/var/www/plugins/CARMACloudPlugin", //path
+                "/bin/CARMACloudPlugin", //exeName
+                "manifest.json", //manifest
+                "500000",//maxMessageInterval
+                "",//commandLineParameters
+            };
+
+            PluginInstallation noInstallation = {
+                -1, //enabled
+                "", //path
+                "", //exeName
+                "", //manifest
+                "",//maxMessageInterval
+                "",//commandLineParameters
+            };
+            
+            telemetryPointer = make_shared<PluginTelemetry>();
+            telemetryPluginInfoOnlyPointer = make_shared<PluginTelemetry>();
+            telemetryPointer->setPluginInfo(pluginInfo);
+            telemetryPluginInfoOnlyPointer->setPluginInfo(pluginInfo);
+            telemetryPointer->setPluginInstallation(installation);
+        }
+        void TearDown() override{}
+    };
+
+    TEST_F(TelemetrySerializerTest, getPluginInfo){
```

Comment:
NIT: This test case and the one below do not test the serializer but test the PluginTelemetry class. This should probably be put into its own unit test class